easybuild/easyblocks/m/mvapich2.py

```diff
@@ -70,12 +70,24 @@ def configure_step(self):
 
         # enable specific support options (if desired)
         if self.cfg['withmpe']:
-            add_configopts.append('--enable-mpe')
+            # --enable-mpe is coming from MPICH.
+            # It is not available anymore in MPICH package since version 3.0, which correspond to MVAPICH2 1.9.
+            # MPE can be downloaded separately at http://www.mpich.org/static/mpe/downloads/
+            # However, the 'withmpe' option should be maintained for backward compatibility purpose
+            if LooseVersion(self.version) < LooseVersion('1.9'):
+                add_configopts.append('--enable-mpe')
+            else:
+                raise EasyBuildError("MPI Parallel Environment (MPE) is not available anymore starting MVAPICH2 1.9")
         if self.cfg['withlimic2']:
             add_configopts.append('--enable-limic2')
         if self.cfg['withchkpt']:
             add_configopts.extend(['--enable-checkpointing', '--with-hydra-ckpointlib=blcr'])
         if self.cfg['withhwloc']:
+            # --with-hwloc/--without-hwloc option is not available anymore MVAPICH2 >= 2.0.
+            # Starting this version, HWLOC is apparently distributed with MVAPICH2 and always compiled with MVAPICH2,
+            # and it cannot be disabled.
+            # The 'withhwloc' option should be maintained for backward compatibility purpose.
+            # EasyBuild and MVAPCH2 will just silently ignore this option if it is used.
             add_configopts.append('--with-hwloc')
```

Comment:
Two things:
- `--with-hwloc` is not recognized by the configure (but it is still in the documentation). I can clarify this with the MVAPICH2 team.
- The `configure` from MPICH/MVAPCH2 is not checking for the unrecognized options. Because they use a top-level configure which is calling many other configure scripts with all the same options (some options being used by some configure scripts, and not by other ones). So #157 will not apply here.